src/test/java/org/elasticsearch/index/query/TemplateQueryParserTest.java

```diff
@@ -122,6 +122,74 @@ public void testParser() throws IOException {
         assertTrue("Parsing template query failed.", query instanceof ConstantScoreQuery);
     }
 
+    /**
+     * Test that the template query parser can parse and evaluate template expressed as a single string
+     * with conditional clause.
+     */
+    @Test
+    public void testParseTemplateAsSingleStringWithConditionalClause() throws IOException {
+        String templateString = "{" +
+                "  \"template\" : \"{ \\\"match_{{#use_it}}{{template}}{{/use_it}}\\\":{} }\"," +
+                "  \"params\":{" +
+                "    \"template\":\"all\"," +
+                "    \"use_it\": true" +
+                "  }" +
+                "}";
+
```

Comment:
> The purpose of this PR was to fix the issue with Search Templates in the first place but IIRC both the Template Query and Search Templates were using the same code for parsing the templates thus I added also tests for Template Query. 

Makes sense. I only saw those and missed your point about the search templates. Should have looked closer at your curl reproduction. Thanks for your clarification.